### PR TITLE
Fixing some issues closes #64, closes #62

### LINKED ISSUE #62: Improve run_all funtion from Experiment.py
2 Things were suggested:
1. Other way of skipping steps - this integer indicator may be very cumbersome in later stages.
~~2. Checking whether the code was changed.~~

Both things should be simple. My suggestion for now is:
1. We iterate over all checks and if all are passed we consider step to be passed.
~~2. As step is being run we save model's code to some hidden folder and calculate it's hash. When step is run the next time. We compare hashes. We show warning to the user that this step should be rerun.~~

If anyone solving this task will have better ideas please suggest them

Acceptance Criteria:
- [x] Checking progress within experiment is no longer done based on integer counter
~~Some primitive method for checking whether code changed and experiment should be done is implemented~~

### LINKED ISSUE #64: get rid of STEPS_REGISTRY
it is introduced in `step.py` and used just by `metric_calculator` in defining exception of when metric shouldn't be calculated. 

The idea beside MetricCalculator is that this is a singleton and user doesn't pass it anywhere. It would require dozen of passes and definitions from the user. 

The only problem is that when we want to define exception (moment when metric shouldn't be calculated) we need list of all STEPS used within this experiment.

My idea is that `Experiment` can pass this to the metric_calculator before we run `run_all` function and then these exceptions are registered.

Acceptance criteria:
- [x] Refactor code so that there is no need of STEPS_REGISTRY and access to steps used within experiments is simpler.

### PR DESCRIPTION
**Pull request by SebChw**
All issues solved beside the one defined on git are marked with **DONE** [here](https://docs.google.com/document/d/1_wVm6b0EWP6oz_vgUiRZiTrs2wrSG8QarnFDa1oOqTg/edit) 

After I obtain LGTM from someone Ill try to adapt art template to it

### CODE REVIEW

File: art/step/step_savers.py [hunk 1]
```diff
@@ -7,28 +7,26 @@
...
         pass
 
     @abstractmethod
     def load(self, step_name: str, filename: str):
         pass
 
-    def ensure_directory(self, step_id: int, step_name: str):
-        return (BASE_PATH / f"{step_id}_{step_name}").mkdir(parents=True, exist_ok=True)
+    def ensure_directory(self, step_id: str, step_name: str):
+        return (BASE_PATH / f"{step_name}_{step_id}").mkdir(parents=True, exist_ok=True)
```
Comment (trebacz626) [inline art/step/step_savers.py]: step_id was before name on purpose so that files would be sorted for user

File: art/step/step_savers.py [hunk 2]
```diff
@@ -7,28 +7,26 @@
...
 
-    def get_path(self, step_id: int, step_name: str, filename: str):
-        return BASE_PATH / f"{step_id}_{step_name}" / filename
+    def get_path(self, step_id: str, step_name: str, filename: str):
+        return BASE_PATH / f"{step_name}_{step_id}" / filename
```
Comment (trebacz626) [inline art/step/step_savers.py]: same here

File: art/step/step.py [hunk 3]
```diff
@@ -36,3 +35,6 @@ def _get_saved_state(self) -> Dict[str, str]:
 
     def get_saved_state(self) -> Dict[str, str]:
         return {}
+
+    def get_step_id(self) -> str:
```
Comment (trebacz626) [inline art/step/step.py:50]: I do not think that steps should have different names based on which datamodule or model was used. Maybe for different model we should just do it in a different folder

File: art/step/step.py [hunk 4]
```diff
@@ -9,21 +9,20 @@
...
 
     def __init__(self):
-        self.STEPS_REGISTRY.append(self)
         self.results = {}
+        self.model = None
+        self.datamodule = None
 
     def __call__(self, previous_states: List[Dict]):
-        ExperimentState.current_stage = TrainingStage.TRAIN # TODO: Some steps need to do this during execution time
+        ExperimentState.current_stage = TrainingStage.TRAIN
```
Comment (trebacz626) [inline art/step/step.py]: I like the stage idea for now, but need to dive deeper.

File: art/step/step_savers.py (hunk 1 above)
Comment (kordc) [reply art/step/step_savers.py]: Sounds reasonable, I'll change that

**Pull request merged**